src/Microsoft.AspNetCore.Mvc.Core/DisableRequestSizeLimitAttribute.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.AspNetCore.Mvc.Core
+{
+    public class DisableRequestSizeLimitAttribute : Attribute, IRequestSizePolicy, IOrderedFilter, IResourceFilter
+    {
+        /// <inheritdoc />
+        public bool IsReusable => true;
+
+        /// <inheritdoc />
+        public int Order { get; set; } = 1000;
+
+        /// <inheritdoc />
+        public void OnResourceExecuted(ResourceExecutedContext context)
+        {
+
+        }
+
+        /// <summary>
+        /// As an <see cref="IResourceFilter"/>, this filter looks at the request and rejects it before going ahead if
+        /// the request body size is greater than the specified limit.
+        /// </summary>
+        /// <param name="context">The <see cref="ResourceExecutingContext"/>.</param>
+        public void OnResourceExecuting(ResourceExecutingContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (IsClosestRequestSizePolicy(context.Filters))
+            {
+                var maxRequestBodySizeFeature = context.HttpContext.Features.Get<IHttpMaxRequestBodySizeFeature>();
+                maxRequestBodySizeFeature.MaxRequestBodySize = null;
```

Comment:
You also need to check IsReadOnly and not try to change the value. Possibly log that the setting could not be applied. This would happen if the request body had already started reading, such as being buffered.
It would also make sense to log if the feature was missing and the attribute couldn't be applied.